### ISSUE TITLE
Log has warning about wrong user casing

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#36139]
### Steps to reproduce
1. send an API request for a user and specify a different mix of case in the UID, e.g. `GET` to an end-point like:
```
ocs/v2.php/cloud/users/BRAND-NEW-USER
```

### Expected behaviour
No warnings in the log.

### Actual behaviour
There is a warning in the log:
```
{"reqId":"pvwkbOcCUj2rfasYDTpy","level":2,"time":"2019-09-03T00:34:08+00:00","remoteAddr":"172.27.0.10","user":"admin","app":"files","method":"GET","url":"\/ocs\/v2.php\/cloud\/users\/BRAND-NEW-USER","message":"initMountPoints() called with wrong user casing. This could be a bug. Expected: \"Brand-New-User\" got \"BRAND-NEW-USER\". Stack: [{\"file\":\"\\\/drone\\\/server\\\/lib\\\/private\\\/Files\\\/Filesystem.php\",\"line\":374,\"function\":\"initMountPoints\",\"class\":\"OC\\\\Files\\\\Filesystem\",\"type\":\"::\"},{\"file\":\"\\\/drone\\\/server\\\/lib\\\/private\\\/legacy\\\/util.php\",\"line\":297,\"function\":\"init\",\"class\":\"OC\\\\Files\\\\Filesystem\",\"type\":\"::\"},{\"file\":\"\\\/drone\\\/server\\\/apps\\\/provisioning_api\\\/lib\\\/Users.php\",\"line\":657,\"function\":\"setupFS\",\"class\":\"OC_Util\",\"type\":\"::\"},{\"file\":\"\\\/drone\\\/server\\\/apps\\\/provisioning_api\\\/lib\\\/Users.php\",\"line\":214,\"function\":\"fillStorageInfo\",\"class\":\"OCA\\\\Provisioning_API\\\\Users\",\"type\":\"->\"},{\"file\":\"\\\/drone\\\/server\\\/lib\\\/private\\\/legacy\\\/api.php\",\"line\":179,\"function\":\"getUser\",\"class\":\"OCA\\\\Provisioning_API\\\\Users\",\"type\":\"->\"},{\"file\":\"\\\/drone\\\/server\\\/lib\\\/private\\\/Route\\\/Router.php\",\"line\":342,\"function\":\"call\",\"class\":\"OC_API\",\"type\":\"::\"},{\"file\":\"\\\/drone\\\/server\\\/ocs\\\/v1.php\",\"line\":63,\"function\":\"match\",\"class\":\"OC\\\\Route\\\\Router\",\"type\":\"->\"},{\"file\":\"\\\/drone\\\/server\\\/ocs\\\/v2.php\",\"line\":23,\"args\":[\"\\\/drone\\\/server\\\/ocs\\\/v1.php\"],\"function\":\"require_once\"}]"} 

```

Actually this changed-case in API requests works fine, you can create the user as any of `Brand-New-User` `brand-new-user1` `BRAND-NEW-USER` and refer to the UID with any other mix of case.

IMO we do not need to emit warnings like this in the log.

### Server configuration
Current core master

**Comment by phil-davis** [comment]
This still happens:

Good example for the exact user name:
```
$ curl -u admin:admin http://172.17.0.1:8080/ocs/v2.php/cloud/users/uu1
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message/>
 </meta>
 <data>
  <enabled>true</enabled>
  <quota>
   <free>63999873024</free>
   <used>82764</used>
   <total>63999955788</total>
   <relative>0</relative>
   <definition>default</definition>
  </quota>
  <email/>
  <displayname>User One</displayname>
  <home>/home/phil/git/owncloud/core/data/uu1</home>
  <two_factor_auth_enabled>false</two_factor_auth_enabled>
 </data>
</ocs>
```

Example using the uppercase of the username:
```
$ curl -u admin:admin http://172.17.0.1:8080/ocs/v2.php/cloud/users/UU1
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message/>
 </meta>
 <data>
  <enabled>true</enabled>
  <quota>
   <definition>default</definition>
  </quota>
  <email/>
  <displayname>User One</displayname>
  <home>/home/phil/git/owncloud/core/data/uu1</home>
  <two_factor_auth_enabled>false</two_factor_auth_enabled>
 </data>
</ocs>
```

owncloud.log from the 2nd example:
```
{"reqId":"hANq1pcYQU5NdvTPXzUf","level":2,"time":"2021-09-21T10:06:09+00:00","remoteAddr":"192.168.1.64","user":"admin","app":"files","method":"GET","url":"\/ocs\/v2.php\/cloud\/users\/UU1","message":"initMountPoints() called with wrong user casing. This could be a bug. Expected: \"uu1\" got \"UU1\". Stack: [{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/lib\\\/private\\\/Files\\\/Filesystem.php\",\"line\":374,\"function\":\"initMountPoints\",\"class\":\"OC\\\\Files\\\\Filesystem\",\"type\":\"::\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/lib\\\/private\\\/legacy\\\/util.php\",\"line\":311,\"function\":\"init\",\"class\":\"OC\\\\Files\\\\Filesystem\",\"type\":\"::\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/apps\\\/provisioning_api\\\/lib\\\/Users.php\",\"line\":677,\"function\":\"setupFS\",\"class\":\"OC_Util\",\"type\":\"::\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/apps\\\/provisioning_api\\\/lib\\\/Users.php\",\"line\":216,\"function\":\"fillStorageInfo\",\"class\":\"OCA\\\\Provisioning_API\\\\Users\",\"type\":\"->\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/lib\\\/private\\\/legacy\\\/api.php\",\"line\":185,\"function\":\"getUser\",\"class\":\"OCA\\\\Provisioning_API\\\\Users\",\"type\":\"->\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/lib\\\/private\\\/Route\\\/Router.php\",\"line\":344,\"function\":\"call\",\"class\":\"OC_API\",\"type\":\"::\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/ocs\\\/v1.php\",\"line\":63,\"function\":\"match\",\"class\":\"OC\\\\Route\\\\Router\",\"type\":\"->\"},{\"file\":\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/ocs\\\/v2.php\",\"line\":23,\"args\":[\"\\\/home\\\/phil\\\/git\\\/owncloud\\\/core\\\/ocs\\\/v1.php\"],\"function\":\"require_once\"}]"}
```

Also note that in the 2nd example the quota details are not in the response.

@JanAckermann Is it easy to fix this little thing?

**Comment by jnweiger** [comment]
confirmed fixed in 10.9.0 beta1
* [x] the level is now 0 (debug), instead of 2 (warn).


* [ ] The quota details are still missing when user name casing is wrong
